integration-tools/src/main/java/uk/ac/bbsrc/tgac/miso/integration/dp5mirage/DP5MirageScanner.java

```diff
@@ -0,0 +1,270 @@
+package uk.ac.bbsrc.tgac.miso.integration.dp5mirage;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpClient.Version;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map.Entry;
+import uk.ac.bbsrc.tgac.miso.integration.BoxScan;
+import uk.ac.bbsrc.tgac.miso.integration.BoxScanner;
+import uk.ac.bbsrc.tgac.miso.integration.util.IntegrationException;
+import java.util.Map;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import uk.ac.bbsrc.tgac.miso.integration.visionmate.VisionMateScan;
+
+
+/**
+ * This class integrates the MirageGP5 scanner with MISO, and involves communicating with
+ * the scanner software via API.
+ * For proper usage, see {@link BoxScanner}
+ */
+public class DP5MirageScanner implements BoxScanner {
+
+  private String[] args;
+  private String host;
+  private int port;
+  private HttpClient httpClient;
+  private int getParamTimeout = 10;
+  private int postActionTimeout = 10;
+  private int putActionTimeout = 10;
+  private int deleteActionTimeout = 10;
+  private String REMOTE_STUB = "/dp5/remote/v1";
+  private HttpRequest request;
+
+  // Setters
+  public void setHost(String host) {
+    this.host = host;
+  }
+
+  public void setPort(int port) {
+    this.port = port;
+  }
+
+  public void setHttpClient(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  public void setGetParamTimeout(int getParamTimeout) {
+    this.getParamTimeout = getParamTimeout;
+  }
+
+  public void setPostActionTimeout(int postActionTimeout) {
+    this.postActionTimeout = postActionTimeout;
+  }
+
+  public void setDeleteActionTimeout(int deleteActionTimeout) {
+    this.deleteActionTimeout = deleteActionTimeout;
+  }
+
+  // Getters
+  public HttpClient getHttpClient() {
+    return this.httpClient;
+  }
+
+  private String getParameterAppend(Map<String,String> parameters) {
+    String appendQuery = "";
+
+    for (Entry<String, String> entry : parameters.entrySet()) {
+      appendQuery = appendQuery + entry.getKey() + "=" + entry.getValue() + "&";
+    }
+    return appendQuery;
```

Comment:
```suggestion
    return parameters.entrySet().stream().map(e -> new StringBuilder(e.getKey()).append("=").append(e.getValue()).toString()).collect(Collectors.joining("&"));
```
(suggestion untested but this would fix the problem of ending with an extra &)